### PR TITLE
Updated Ronn - use-case example

### PR DESCRIPTION
**Pull request by WestleyK**
A couple things about the PR:
I'm not to thrilled about the file-generating directory that ronn is in, like it could have a better name.

Also, I'm not sure about the asciinema demo, I'll be fine taking it down if you don't like it, Or I can spend more time on it and make it better if you think its worth it.

And yes, I know, there are a lot of commits, I'm still figuring this github branch thing...

The definition file is very big, It could be cleaned up a lot, But it does work very well, no bugs in the shell part (that I can find... yet).

### CODE REVIEW

File: file-generating/ronn/README.md
Comment (jchavezlavalle) [inline file-generating/ronn/README.md]: change heres -> here's

File: file-generating/ronn/README.md
Comment (jchavezlavalle) [inline file-generating/ronn/README.md]: Then, add it to your PATH ...

File: file-generating/ronn/README.md
Comment (jchavezlavalle) [inline file-generating/ronn/README.md]: change let to "let's" or "we will "

File: file-generating/ronn/README.md
Comment (jchavezlavalle) [inline file-generating/ronn/README.md]: change Lets to Let's

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: guess it's **process**?

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: check if file **exists**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: change smthing to something and happend to happened

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: change extention to extension

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: again change exist to exists

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: Already **exists**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: check **if** TMP_FILE **exists**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: Already **exists**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: typo in "Already" and also change exist to exists

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: check if the HTML file **exists**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: Already **exists**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: Typo in **generating** and manpage should be with no caps

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: again exist -> exists

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: typo in word Already, also exist -> **exists**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: typo in Already, change exist -> exists

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: Typo: more argument **processing**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: maybe add a verb in here: "**verify** which files exist for output message" for example

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: **making** the help file..

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: specify **an** output file

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: **making** a manpage:...

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: 255 - **something** very bad **happened**

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: makeing -> making

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: ronn **license** on American English I guess? :sweat_smile:

File: file-generating/ronn/ronn.sif.md
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.sif.md]: would uppercase Alpine and also Ronn: -> Ronn in an Alpine container . Also, is the **(1)** necessary? otherwise maybe it's better to be removed

File: file-generating/ronn/ronn.sif.md
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.sif.md]: Ronn (v0.7.3) **in an** Alpine (3.8) container..

File: file-generating/ronn/ronn.sif.md
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.sif.md]: specify **an** output file

File: file-generating/ronn/ronn.sif.md
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.sif.md]: Makeing -> making

File: file-generating/ronn/ronn.sif.md
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.sif.md:77]: Typo on title: Auther -> Author

File: file-generating/ronn/template.md
Comment (jchavezlavalle) [inline file-generating/ronn/template.md]: typo in program, also remove (1)

File: file-generating/ronn/template.md
Comment (jchavezlavalle) [inline file-generating/ronn/template.md]: example of **something**

File: file-generating/ronn/template.md
Comment (jchavezlavalle) [inline file-generating/ronn/template.md]: typo in something

File: file-generating/ronn/template.md
Comment (jchavezlavalle) [inline file-generating/ronn/template.md]: typo in license

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]
Some typos to be fixed :+1: @WestleyK

File: file-generating/ronn/ronn.sif.md
Comment (WestleyK) [reply file-generating/ronn/ronn.sif.md]: The `(1)` is not really necessary, but it shows the user that they can replace the `1` with other numbers for different manpage types, like `1` = `Executable programs or shell commands`, and `6` = `Games` (for game manuals), etc...

File: file-generating/ronn/template.md
Comment (WestleyK) [reply file-generating/ronn/template.md]: The `(1)` is not really necessary, but it shows the user that they can replace the `1` with other numbers for different manpage types, like `1` = `Executable programs or shell commands`, and `6` = `Games` (for game manuals), etc...

**Review by jchavezlavalle — APPROVED** [review]

**Comment by jchavezlavalle** [comment]
Please make sure to rebase this PR @WestleyK

**Comment by jchavezlavalle** [comment]
> Please make sure to rebase this PR @WestleyK

looks good to go, could you rebase this PR? @WestleyK

File: file-generating/ronn/ronn.def
Comment (jmstover) [inline file-generating/ronn/ronn.def]: `making`

File: file-generating/ronn/template.md
Comment (jchavezlavalle) [inline file-generating/ronn/template.md]: change to "author"

File: file-generating/ronn/ronn.def
Comment (jchavezlavalle) [inline file-generating/ronn/ronn.def]: or already **exists**

File: file-generating/ronn/README.md
Comment (jchavezlavalle) [inline file-generating/ronn/README.md:143]: This is a dead link

File: file-generating/ronn/README.md
Comment (jchavezlavalle) [inline file-generating/ronn/README.md:97]: check this url as well

**Review by jchavezlavalle — COMMENTED** [review]
uh-oh, just noticed some small typos around, no big deal @WestleyK after that I'll approve :+1:

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]
check previous review :top:

File: file-generating/ronn/README.md
Comment (WestleyK) [reply file-generating/ronn/README.md:97]: The url seems fine... once it is merged into sylabs.

File: file-generating/ronn/README.md
Comment (WestleyK) [reply file-generating/ronn/README.md:143]: Well, if you replace `sylabs` with `WestleyK`, and `master` to `WestleyK-ronn`, it seems to work fine for my branch, so I think it will work once it is merged... I think...

**Review by jchavezlavalle — APPROVED** [review]

File: file-generating/ronn/ronn.def
Comment (jmstover) [inline file-generating/ronn/ronn.def:57]: Where are you getting OPTION from? Below you're doing:
```
for OPTION in "$@"; do
  check_args
done
```
But you aren't _passing_ OPTION ... and it _should_ be out of scope. 

IMO you should do:
```
check_args $OPTION
```

And in check_args do:
```
check_args() {
    OPTION = $1
    [ ... ]
```

**Review by jmstover — CHANGES_REQUESTED** [review]

File: file-generating/ronn/ronn.def
Comment (WestleyK) [reply file-generating/ronn/ronn.def:57]: Yeah! thats what I meant to do.
Or we could put the whole `check_args()` function in the `for` loop, that may be a better idea... 

Again, I didin't spend very much time on this example, This was a "learn something cool instead of watching a movie", this was for my entertainment not really for usefulness.

Comment (jmstover) [reply file-generating/ronn/ronn.def:57]: Nah... I would leave `check_args()` as it is, and just pass the value to it. Less changes, and it's a bit more proper.

**Review by jmstover — APPROVED** [review]

**Comment by WestleyK** [comment]
Wait! don't merge it yet!

**Comment by WestleyK** [comment]
Small change, almost ready!

**Comment by jmstover** [comment]
I was actually looking to message you and see if it was good to merge. ;)

**Comment by WestleyK** [comment]
Almost! LOL

File: file-generating/ronn/README.md
Comment (jmstover) [inline file-generating/ronn/README.md]: Change `function` to be `option`

**Review by jmstover — CHANGES_REQUESTED** [review]

**Comment by WestleyK** [comment]
Okay! Now I think its ready!

**Review by jmstover — APPROVED** [review]

**Pull request merged**
